components/agent-version-display.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import React, { useCallback, useEffect, useState } from "react";
+
+interface AgentVersionDisplayProps {
+  instanceId: string;
+  publicDns: string;
+  keyName: string;
+  instanceState: string;
+}
+
+const AgentVersionDisplay = ({
+  instanceId,
+  publicDns,
+  keyName,
+  instanceState,
+}: AgentVersionDisplayProps) => {
+  const [version, setVersion] = useState<string>("Loading...");
+
+  const fetchVersion = useCallback(async () => {
+    if (instanceState !== "running") {
+      setVersion("N/A");
+      return;
+    }
+
+    try {
+      const response = await fetch("/api/health-check", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ instanceId, publicDns, keyName }),
+      });
```

Comment:
Aren't we already calling the health check api on this instance page? If so, hoist and consolidate the logic, passing down the results to the relevant components